### DISCUSSION TITLE
react re-render fire just init time

### DISCUSSION THREAD

**Discussion by gseok** [pmndrs/jotai#2383] [Bug report]
## Summary

when I make very simple component and check re-render.
that time, if using jotai(useAtomValue) then always re-render..
is it jotai's spec??

I'm try to very simple follow code

```ts
const countAtom = atom(0);

// jotai's doc simple counter component
export const Counter = () => {
  const setCount = useSetAtom(countAtom);
  const count = useAtomValue(countAtom);

  console.log('Counter component re-render');

  return (
    <>
      <div>count: {count}</div>
      <button onClick={() => setCount(count + 1)}>+1</button>
    </>
  );
};

// just my app simple Count using
const App = () => {
  return <Counter />;
}
```
- That is very simple logic But.. `always ('Counter component re-render') log printed twice!! ` even if button not clicked!
- jotais' follow logic always fire re-render.. (Line 97)
    - https://github.com/pmndrs/jotai/blob/main/src/react/useAtomValue.ts#L97
- I'm just init and open and.. there are no changed.!!  but alway re-render twice.. Is is normal operation???
    - if remove code(https://github.com/pmndrs/jotai/blob/main/src/react/useAtomValue.ts#L97) then just one render..

so i'm report these problem.

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by gseok** [comment]
additional info. 

- Not used 'strict mode' 
- But always re-render twice init time

**Comment by dai-shi** [comment]
It's an unavoidable / intentional behavior with the current implementation (based on `useReducer` and no `useRef` hacks.)
#1444 
#1015
